index.js

```diff
@@ -1,7 +1,8 @@
-module.exports = function (main) {
-  main = main || require.main
-  if (isIISNode(main)) return handleIISNode(main)
-  else return main.filename
+module.exports = function (_require) {
+  _require = _require || require
+  var main = _require.main
+  if (main && isIISNode(main)) return handleIISNode(main)
+  else return main ? main.filename : process.cwd()
```

Comment:
yeah, I think we're probably cool; my feeling is that returning the folder is the correct behavior, here's a really simple reproduction I created:

**before:**

```sh
Benjamins-MacBook-Pro:tmp benjamincoe$ node
> require('require-main-filename')()
TypeError: Cannot read property 'filename' of undefined
    at isIISNode (/private/tmp/node_modules/require-main-filename/index.js:8:33)
    at module.exports (/private/tmp/node_modules/require-main-filename/index.js:3:7)
    at repl:1:33
    at REPLServer.defaultEval (repl.js:164:27)
    at bound (domain.js:250:14)
    at REPLServer.runBound [as eval] (domain.js:263:12)
    at REPLServer.<anonymous> (repl.js:392:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:210:10)
> 
```

**after:**

```sh
Benjamins-MacBook-Pro:tmp benjamincoe$ node
> require('require-main-filename')()
'/private/tmp'
```